test/Fortran/USE-MODULE-live.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+#
+# MIT License
+#
+# Copyright The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+Test Fortran module use, organized in such a way that
+module files will be "not found" unless dependencies have been forced.
+See PR #4452.
+
+This is a "live" test - uses an actual Fortran compiler, as the
+mock compiler would need to have a sophisticated regex to catch
+the situation. Possibly we could convert later if the mock fortran.py
+gets smarter.
+"""
+
+import TestSCons
+
+_python_ = TestSCons._python_
+_exe = TestSCons._exe
+
+test = TestSCons.TestSCons()
+
```

Comment:
Probably need to gate this test on finding gfortran?

Reply:
I can reproduce now after adding the mingw64 install to my PATH, which I intentionally hadn't done before.  

I should add the gfortran gate, but it won't help the failure on AppVeyor (or the one I now have), where gfortran *is* being found.  The `mslink` tool has still been run, and it's set this:
```py
exeLinkAction = SCons.Action.Action('${TEMPFILE("$LINK $LINKFLAGS /OUT:$TARGET.windows $_LIBDIRFLAGS $_LIBFLAGS $_PDB $SOURCES.windows", "$LINKCOMSTR")}', '$LINKCOMSTR')
```
which is the wrong syntax if using `gfortran` as the linker (to be more precise, as the dispatcher to the linker). Adding gnulink to the tools list makes it work for me, that seems to be kind of forcing the solution, but...it's a testcase.

Reply:
Yup. gate for gfortran, and then only load gfortran and gnulink in the SConstruct, and that shouldn't fail for environmental reasons.. Might make sense to change LINK = "$FORTRAN" ?